numpy/core/shape_base.py

```diff
@@ -345,7 +345,7 @@ def hstack(tup):
         return _nx.concatenate(arrs, 1)
 
 
-def _stack_dispatcher(arrays, axis=None, out=None):
+def _stack_dispatcher(arrays, axis=None, out=None, casting=None):
```

Comment:
This and everything else should be keyword argument only, so `, *, casting=None`.

Reply:
Hello @seberg we added the option casting in the recent commits